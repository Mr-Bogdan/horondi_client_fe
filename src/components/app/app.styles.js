import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles(() => ({
  app: {
    overflowY: 'scroll',
  },
  center: {
    width: '3rem',
    margin: '22rem auto',
    '@media (max-width: 1400px)': {
      margin: '13rem auto'
    }
  }
}));
