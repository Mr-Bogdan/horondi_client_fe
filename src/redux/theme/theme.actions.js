import { SET_FILTER_MENU_STATUS } from './theme.types';

export const setFilterMenuStatus = (newFilterMenuStatus) => ({
  type: SET_FILTER_MENU_STATUS,
  payload: newFilterMenuStatus
});
