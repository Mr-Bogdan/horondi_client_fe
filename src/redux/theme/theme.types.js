export const SET_THEME_MODE = 'SET_THEME_MODE';
export const SET_FILTER_MENU_STATUS = 'SET_FILTER_MENU_STATUS';
