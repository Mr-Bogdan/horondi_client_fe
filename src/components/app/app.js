import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { ThemeProvider } from '@material-ui/styles';
import { CssBaseline } from '@material-ui/core';
import ThemeContext from '../../context/theme-context';
import Routes from '../../routes';
import Chat from '../../containers/chat';
import { theme } from './app-theme/app.theme';
import { LIGHT_THEME } from '../../configs';
import { useStyles } from './app.styles';
import { getFromLocalStorage } from '../../services/local-storage.service';
import { getCategories } from '../../redux/categories/categories.actions';
import { preserveUser } from '../../redux/user/user.actions';
import { getContacts } from '../../redux/contacts/contacts.actions';
import { selectLocation } from '../../redux/selectors/multiple.selectors';

const App = () => {
  const [appTheme, setAppTheme] = useState(true);
  const { location } = useSelector(selectLocation);
  const dispatch = useDispatch();
  const styles = useStyles({ isHome: location === '/' });

  let localStorageThemeMode = getFromLocalStorage('theme');
  const themeMode = localStorageThemeMode === LIGHT_THEME;
  if (!localStorageThemeMode) {
    localStorageThemeMode = LIGHT_THEME;
  }
  const themeValue = theme(localStorageThemeMode);

  useEffect(() => {
    dispatch(preserveUser());
    dispatch(getCategories());
    dispatch(getContacts());
  }, []);

  useEffect(() => {
    setAppTheme(themeMode);
  });

  return (
    <div className={styles.mainBar}>
      <ThemeProvider theme={themeValue}>
        <ThemeContext.Provider value={[appTheme, setAppTheme]}>
          <CssBaseline />
          <Routes />
          <Chat />
        </ThemeContext.Provider>
      </ThemeProvider>
    </div>
  );
};

export default App;
