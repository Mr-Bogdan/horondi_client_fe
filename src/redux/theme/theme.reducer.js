import { SET_THEME_MODE } from './theme.types';

const initialState = {
  lightMode: true
};

const themeState = (state = initialState, { type, payload } = {}) => {
  if (type === SET_THEME_MODE) {
    return {
      ...state,
      lightMode: payload
    };
  } else {
    return state;
  }
};

export default themeState;
