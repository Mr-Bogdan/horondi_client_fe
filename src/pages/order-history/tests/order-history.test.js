import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRedux from 'react-redux';
import { ThemeProvider } from '@material-ui/styles';
import { BrowserRouter } from 'react-router-dom';
import OrderHistory from '../order-history';
import { theme } from '../../../components/app/app-theme/app.theme';
import mockStore from './mockStore';

Enzyme.configure({ adapter: new Adapter() });

const themeValue = theme('light');
let spyOnUseSelector;
let spyOnUseDispatch;
let mockDispatch;
let wrapper;

describe('Recovery component tests', () => {
  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(reactRedux, 'useSelector');
    spyOnUseDispatch = jest.spyOn(reactRedux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseSelector.mockImplementation(() => mockStore);
    spyOnUseDispatch.mockReturnValue(mockDispatch);

    wrapper = shallow(
      <BrowserRouter>
        <ThemeProvider theme={themeValue}>
          <OrderHistory />
        </ThemeProvider>
      </BrowserRouter>
    );
  });

  afterEach(() => {
    jest.restoreAllMocks();
    spyOnUseSelector.mockClear();
    wrapper = null;
  });

  it('Should render Recovery', () => {
    expect(wrapper).toBeDefined();
  });
});
