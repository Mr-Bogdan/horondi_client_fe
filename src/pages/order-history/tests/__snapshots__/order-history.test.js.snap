// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 404 page 1`] = `undefined`;
